### DISCUSSION TITLE
Volume of wav's in playlist jump wildly from track to track

### DISCUSSION THREAD

**Discussion by ACiDGRiM** [music-assistant/support#1978] [Q&A] [answered]
I'm using chat gpt to read my son a bedtime story, and the issue is that when a long 8 paragraph story is sent to the homeassistant tts thru piper, it will be unreliable if it returns a wave.
So on my quest to resolve this issue I came across music assistant, because I chop the story up into 12 word requests to the tts api (making sure to end on a period) and then combining the url's in a playlist. at first I was just playing to sonos directly which had consistent volume, but was unreliably in the right order. Musicassistant solved this by re-streaming the playlist in the order of the string or comma seperated tts url's.

Unfortunately, this caused another issue, the volume between each "track" is either just right or suddenly loud, followed by just right. So while the "computer mom" is softly reading the story of Sir <child's name> suddenly she's yelling about toad stools and the brave little rabbit, and then back to softly reading about the firey dragon and Sir <child's name> fighting to the death.
Is there a feature I'm missing to level the tracks, even if it requires additional pre-processing before the stream begins?

**Comment by OzGav** [comment]
I'm not clear on where the tracks are coming from?

**Reply by ACiDGRiM** [reply]
They are wavs from the Wyoming piper container/addon in homeassistant. I'm sending a comma separated string with the wav urls in order to music assistant.

**Reply by OzGav** [reply]
OK if you are doing this via TTS then that is not implemented yet so that is a problem for a start

**Reply by ACiDGRiM** [reply]
No im using the homeassistant api for tts.
All music assistant gets is a list of wav to play, no different than music
The issue is that the track volume between each track is not normalized

**Reply by OzGav** [reply]
As I said TTS is not implemented yet so weirdness is to be expected. I will make a note against the TTS task though to consider whether multiple TTS messages can be nornalised...

**Reply by ACiDGRiM** [reply]
I see you mean it's as a unique audio signature. Thought you meant it's from musicassistant itself. Thanks for considering it.

**Comment by marcelveldt** [comment, marked as answer]
This should be fixed in recent betas of Music Assistant, can you confirm ?

**Reply by ACiDGRiM** [reply]
It seems to be working much better.  Thanks a lot
